### ISSUE TITLE
Feature request: add warning mode

### ISSUE THREAD

**Issue by noveogroup-amorgunov** [feature-sliced/steiger#34]
Hey. I have a feature request: I want to see errors in warning mode (like in eslint). This will allow you to see errors, but not crash in CI.

It could look like:

```ts
export default defineConfig({
  rules: {
    'forbidden-imports': 'warn'
  },
})
```

**Comment by illright** [comment]
Hi! I'm a bit hesitant to go forward with this, mostly after discovering the project https://github.com/bfanger/eslint-plugin-only-warn and reading its motivation in the readme. I've grown fond of the idea that all issues are issues, and if something is not an issue, it shouldn't be checked for. 

What is your specific use case that would warrant having a warning as a diagnostic type?

**Comment by saaaaaaaaasha** [comment]
> What is your specific use case that would warrant having a warning as a diagnostic type?


@illright if u start new project, its good approach. But if I integrate FSD to existed project (or for example I want use steiger in project with FSD but have a lot of problems), I want to use incremental way.

And I want see warnings in CI

**Comment by noveogroup-amorgunov** [comment]
> What is your specific use case that would warrant having a warning as a diagnostic type?

@illright if u start new project, its good approach. But if I integrate FSD to existed project (or for example I want use steiger in project with FSD but have a lot of problems), I want to use incremental way.

And I want see warnings in CI

**Comment by illright** [comment]
Tracked (and implemented) in #70
